common/functions/src/scalars/maths/trigonometric.rs

```diff
@@ -60,29 +69,96 @@ impl Function for TrigonometricFunction {
+        if is_numeric(&args[0]) || args[0] == DataType::String || args[0] == DataType::Null {
+            Ok(DataType::Float64)
+        } else {
+            Err(ErrorCode::IllegalDataType(format!(
+                "Expected numeric, but got {}",
+                args[0]
+            )))
+        }
     }
 
     fn nullable(&self, _input_schema: &DataSchema) -> Result<bool> {
-        Ok(false)
+        Ok(true)
     }
 
     fn eval(&self, columns: &DataColumnsWithField, _input_rows: usize) -> Result<DataColumn> {
-        let result = columns[0]
-            .column()
-            .to_minimal_array()?
-            .cast_with_type(&DataType::Float64)?
-            .f64()?
-            .apply_cast_numeric(|v| match self.t {
-                Trigonometric::COS => v.cos(),
-                Trigonometric::SIN => v.sin(),
-                Trigonometric::COT => 1.0 / v.tan(),
-                Trigonometric::TAN => v.tan(),
+        let result = if columns.len() == 1 {
+            let opt_iter = columns[0]
+                .column()
+                .to_minimal_array()?
+                .cast_with_type(&DataType::Float64)?;
+            let opt_iter = opt_iter.f64()?.into_iter().map(|v| match v {
+                Some(&v) => match self.t {
+                    Trigonometric::COS => Some(v.cos()),
+                    Trigonometric::SIN => Some(v.sin()),
+                    Trigonometric::COT => Some(1.0 / v.tan()),
+                    Trigonometric::TAN => Some(v.tan()),
+                    Trigonometric::ACOS => {
+                        if (-1_f64..=1_f64).contains(&v) {
+                            Some(v.acos())
+                        } else {
+                            None
+                        }
+                    }
+                    Trigonometric::ASIN => {
+                        if (-1_f64..=1_f64).contains(&v) {
+                            Some(v.asin())
+                        } else {
+                            None
+                        }
+                    }
+                    Trigonometric::ATAN => Some(v.atan()),
+                    _ => unreachable!(),
+                },
+                None => None,
             });
+            DFFloat64Array::new_from_opt_iter(opt_iter)
+        } else {
+            let y_column: &DataColumn = &columns[0].column().cast_with_type(&DataType::Float64)?;
+            let x_column: &DataColumn = &columns[1].column().cast_with_type(&DataType::Float64)?;
+
+            match (y_column, x_column) {
+                (DataColumn::Array(y_series), DataColumn::Constant(x, _)) => {
+                    let x = DFTryFrom::try_from(x.clone())?;
+                    let opt_iter = y_series.f64()?.into_iter().map(|y| y.map(|y| y.atan2(x)));
+                    DFFloat64Array::new_from_opt_iter(opt_iter)
+                }
+                (DataColumn::Constant(y, _), DataColumn::Array(x_series)) => {
+                    let y: f64 = DFTryFrom::try_from(y.clone())?;
```

Comment:
If `y` is None, the result should be None.

Reply:
You are right. I missed this. > <